File: Tests/SimpleTestCases/Constructors.cs
using System;

public static class Program { 
    public static void Main (string[] args) {
        Console.WriteLine(new CustomType());
    }
}

public class CustomType {
    public static int A;
    public int B;
    
    static CustomType () {
        A = 1;
    }
  
    public CustomType () {
        B = 2;
    }
    
    public override string ToString () {
        return String.Format("A = {0}, B = {1}", A, B);
    }
}

File: Tests/SimpleTestCases/Constructors.vb
Imports System

Public Module Program
    Public Sub Main(ByVal Args As String())
        Console.WriteLine(New CustomType())
    End Sub
End Module

Public Class CustomType
    Public Shared A As Integer
    Public B As Integer

    Shared Sub New()
        A = 1
    End Sub

    Public Sub New()
        B = 1
    End Sub

    Public Overrides Function ToString() As String
        Return String.Format("A = {0}, B = {1}", A, B)
    End Function
End Class